exporter/prometheusexporter/prometheus.go

```diff
@@ -63,11 +63,14 @@ func (pe *prometheusExporter) Start(ctx context.Context, host component.Host) er
 		return err
 	}
 
-	pe.shutdownFunc = ln.Close
-
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", pe.handler)
 	srv, err := pe.config.ToServer(ctx, host, pe.settings, mux)
+	pe.shutdownFunc = func(ctx context.Context) error {
+		err := ln.Close()
+		err2 := srv.Shutdown(ctx)
```

Comment:
will srv be nil if ToServer returned an error above?

Reply:
You're right, I'll adjust it accordingly!

Reply:
I added the nil check in the shutdown handler and kept it where it currently is. Alternatively, we could reassign the shutdownFunc different implementations, but personally I find that more confusing.

Reply:
this looks good to me